packages/algob/src/lib/tx.ts

```diff
@@ -266,20 +268,70 @@ export async function makeAndSignTx (
   return [txns, signedTxn];
 }
 
+/**
+ * Signs transaction object(s) and returns raw signed transaction
+ * Note: Sign transaction is used to sign single transaction and `signTransactions` takes
+ * SDK transaction object, signer and signs it.
+ * @param txnAndSign Transaction object(s) with signers
+ */
+export function signTransactions (
+  txnAndSign: wtypes.TransactionAndSign[]
+): Uint8Array | Uint8Array[] {
+  let txns: Transaction[] = [];
+  const signers: wtypes.Sign[] = [];
+  txnAndSign.forEach(function (value, idx) {
+    txns[idx] = value.transaction;
+    signers[idx] = value.sign;
+  });
+  txns = algosdk.assignGroupID(txns);
+  return txns.map((txn: Transaction, index: number) => {
+    return signTransaction(txn, signers[index]);
+  });
+}
+
 /**
  * Execute single transaction or group of transactions (atomic transaction)
+ * executes `ExecParams` or `Transaction` Object, SDK Transaction object passed to this function
+ * will be signed and sent to network. User can use SDK functions to create transactions.
+ * Note: If passing transaction object a signer/s must be provided.
  * @param deployer Deployer
- * @param execParams transaction parameters or atomic transaction parameters
+ * @param transactionParam transaction parameters or atomic transaction parameters
  * https://github.com/scale-it/algo-builder/blob/docs/docs/guide/execute-transaction.md
+ * or TransactionAndSign object(SDK transaction object and signer parameters)
  */
 export async function executeTransaction (
   deployer: Deployer,
-  execParams: wtypes.ExecParams | wtypes.ExecParams[]):
-  Promise<ConfirmedTxInfo> {
-  deployer.assertCPNotDeleted(execParams);
+  transactionParam: (wtypes.ExecParams | wtypes.TransactionAndSign)
+  | (wtypes.ExecParams[] | wtypes.TransactionAndSign[])
+): Promise<ConfirmedTxInfo> {
+  if (Array.isArray(transactionParam)) {
+    if (transactionParam.length === 0) {
+      throw new BuilderError(ERRORS.GENERAL.EXECPARAMS_LENGTH_ERROR);
+    }
+    if (wtypes.isSDKTransactionAndSign(transactionParam[0])) {
+      const signedTxn = signTransactions(transactionParam as wtypes.TransactionAndSign[]);
+      const confirmedTx = await deployer.sendAndWait(signedTxn);
+      console.log(confirmedTx);
+      return confirmedTx;
+    }
+  } else {
+    if (wtypes.isSDKTransactionAndSign(transactionParam)) {
+      const signedTxn = signTransaction(transactionParam.transaction, transactionParam.sign);
+      const confirmedTx = await deployer.sendAndWait(signedTxn);
+      console.log(confirmedTx);
+      return confirmedTx;
```

Comment:
we duplicate code in the conditions above. let's try to make a condition which will allow us to not duplicate the code, eg:
```
let isArray = ...

if isArray and length == 0 { throw exception }
if (isArray and isSDKTransactionAndSign(transactions[0])) || (!isArray and isSDKTransactionAndSign(transactions[0])) { ... }

Reply:
ok will update the first condition, but there are different functions used in both the conditions `signTransaction` `signTransactions`.

Reply:
Moved common code after condition checks

Reply:
ok